climada/util/yearsets.py

```diff
@@ -0,0 +1,266 @@
+"""
+This file is part of CLIMADA.
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+---
+Define functions to handle impact year_sets
+"""
+
+import copy
+import logging
+import numpy as np
+from numpy.random import default_rng
+
+import climada.util.dates_times as u_dt
+
+LOGGER = logging.getLogger(__name__)
+
+def impact_yearset(eis, n_resampled_years=None, year_list=None, multiple_events=True,
+                   sampling_vect=None, nr_events_per_year=None):
+
+    """PURPOSE:
+      convert an event (per occurrence) impact set (eis) into an annual impact
```

Comment:
By 'eis', do you simply mean an Impact() object? 

Currently this is not very clear. In CLIMADA, there is no concept of event impact set and annual impact set in the core engine. It is not a good idea to introduce a new nomenclature in the middle of the code.

Maybe something like this is clearer? 
```suggestion
      Create an annual impact set (ais) by sampling events for each year from an existing probabilistic event impact set (eis).
```